maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java

```diff
@@ -256,8 +256,7 @@ public CLIManager() {
                 .build());
         options.addOption(Option.builder(LEGACY_LOCAL_REPOSITORY)
                 .longOpt("legacy-local-repository")
-                .desc(
-                        "Use Maven 2 Legacy Local Repository behaviour, ie no use of _remote.repositories. Can also be activated by using -Dmaven.legacyLocalRepo=true")
+                .desc("Ineffective, only kept for backward compatibility.")
```

Comment:
If such option will have  no effect - why we want to preserve it ... 
If user use it - it will not works as user expect - so we will have silently different behavior that user think

For me will be better to drop at all -  it will more clear for user and do not make confusing

Reply:
Agreed, i was drawn by other left defunct options...